internal/provider/role_mapping_aws.go

```diff
@@ -0,0 +1,189 @@
+package provider
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/tabular-io/tabular-sdk-go/tabular"
+	"github.com/tabular-io/terraform-provider-tabular/internal/provider/util"
+)
+
+var (
+	_ resource.Resource                = &roleMappingAWSResource{}
+	_ resource.ResourceWithConfigure   = &roleMappingAWSResource{}
+	_ resource.ResourceWithImportState = &roleMappingAWSResource{}
+)
+
+type roleMappingAWSResource struct {
+	client *util.Client
+}
+
+func NewRoleMappingAWSResource() resource.Resource {
+	return &roleMappingAWSResource{}
+}
+
+type roleMappingAWSResourceModel struct {
+	Id               types.String `tfsdk:"id"`
+	Name             types.String `tfsdk:"name"`
+	RoleId           types.String `tfsdk:"role_id"`
+	CredentialKey    types.String `tfsdk:"credential_key"`
+	CredentialSecret types.String `tfsdk:"credential_secret"`
+	AWSRoleArn       types.String `tfsdk:"aws_role_arn"`
+}
+
+func (r *roleMappingAWSResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*util.Client)
+}
+
+func (r *roleMappingAWSResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_role_mapping_aws"
+}
+
+func (r *roleMappingAWSResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Tabular Service Account",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Credential ID",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "Service account name",
+				Required:    true,
+			},
+			"role_id": schema.StringAttribute{
+				Description: "Role ID",
+				Required:    true,
+			},
+			"credential_key": schema.StringAttribute{
```

Comment:
Do we want to expose/is there any value to exposing that these are actually credentials under the covers?

Reply:
It's probably not necessary in this case. The credential key is a good choice for the `id` attribute, but it probably doesn't need to be accessed outside of that.